File: 프로그래머스/2/131127. 할인 행사/할인 행사.js
function solution(want, number, discount) {
    let answer = 0;
    
    const wantMap = {};
    want.forEach((item, idx) => {
        wantMap[item] = number[idx];
    });

    for (let i = 0; i <= discount.length - 10; i++) {
        const currentDiscount = {};
        
        for (let j = i; j < i + 10; j++) {
            const item = discount[j];
            currentDiscount[item] = (currentDiscount[item] || 0) + 1;
        }

        if (isMatch(wantMap, currentDiscount)) {
            answer++;
        }
    }

    return answer;
}

function isMatch(obj1, obj2) {
    for (const key in obj1) {
        if (obj1[key] !== obj2[key]) {
            return false;
        }
    }
    return true;
}

File: 프로그래머스/2/131127. 할인 행사/할인 행사.py
from collections import Counter

def solution(want, number, discount):
    want_dict = {w: n for w, n in zip(want, number)}
    answer = 0

    for i in range(len(discount) - 9):
        current_10_days = Counter(discount[i : i + 10])
        
        if current_10_days == want_dict:
            answer += 1
            
    return answer